ooniprobe/View/TestResults/Footer/UploadFooterViewController.m

```diff
@@ -79,22 +81,38 @@ -(void)uploadMeasurements:(NSArray *)notUploaded{
             hud.bezelView.color = [UIColor lightGrayColor];
             hud.backgroundView.style = UIBlurEffectStyleRegular;
         });
-        logs = [NSMutableArray new];
         NSUInteger errors = 0;
         if ([notUploaded count] == 0) return;
         NSUInteger i = 0;
         float progress = 0.0f;
         float measurementValue = 1.0/([notUploaded count]);
-        id<CollectorTask> task = [Engine collectorTaskWithSoftwareName:SOFTWARE_NAME
-                               softwareVersion:[VersionUtility get_software_version]];
+        NSError *error;
+        PESession* session = [[PESession alloc] initWithConfig:
+                              [Engine getDefaultSessionConfigWithSoftwareName:SOFTWARE_NAME
+                                                              softwareVersion:[VersionUtility get_software_version]
+                                                                       logger:self.logger]
+                                                                        error:&error];
+        if (error != nil) {
+            //TODO Do we care about the error?
+        }
+        // Updating resources with no timeout because we don't know for sure how much
+        // it will take to download them and choosing a timeout may prevent the operation
+        // to ever complete. (Ideally the user should be able to interrupt the process
+        // and there should be no timeout here.)
+        [session maybeUpdateResources:[session newContext]
+                                error:&error];
+        //TODO what to do with error?
+        if (error != nil) {
+            //TODO Do we care about the error?
```

Comment:
```suggestion
            //TODO Do we care about the error?
            return;
```

Reply:
return in the first if or second?

Reply:
I think we should return in any case there is an error.

Reply:
even for `maybeUpdateResources` ? is it a problem running a test with old resources?

Reply:
Yes, I think we should return. There are at least reasons for this:

1. the accuracy of measurements depends on the resources being up to date (and we have had data quality incidents in the past caused by assets not being up-to-date;

2. if you eventually ignore the error, the code in probe-engine that will execute later will just try again to fetch the resources and fail, so there is no easy escape from fetching the resources;

3. while it could be tempting to rely on the fact that probe-engine would try to fetch the resources, I'd rather not do that, because that would lead to context mismatches (e.g. we have a 30 seconds deadline for finding the location and, if we use such deadline for possibly downloading the resources and finding the location, certain clients may loop because they cannot really download the resources in 30 seconds).